dotcom-rendering/src/model/enhanceCollections.ts

```diff
@@ -0,0 +1,40 @@
+import { ArticleDesign } from '@guardian/libs';
+import { decideFormat } from '../web/lib/decideFormat';
+
+const enhanceCards = (collections: FEFrontCard[]): DCRFrontCard[] => {
+	const enhanced: DCRFrontCard[] = [];
+	collections.forEach((faciaCard) => {
```

Comment:
Is there a reason we don't use map as we do below?

Reply:
Because map will return something for _all_ items but in this case we might want to filter things if the data isn't available.

Reply:
I think it’s best to keep to one style, either mutation with a `for … of` loop, or functional with a `.filter().map()`. 
Sadly, `forEach` is this weird in-between…

Reply:
I did try `filter` but I was still getting typescript errors 😢 

In my mind (with my personal mental models) I would say both `map` and `forEach` are valid options with use cases where both are valid but also use cases where one or the other might be better suited.

Reply:
That said, I'm totally happy to forgo `forEach` in favour of `for ... of` - sorry, I didn't speak to that in my last comment

Reply:
subjectively `for ... of` is really readable.

Reply:
What were the typescript errors?

Reply:
> Because map will return something for _all_ items but in this case we might want to filter things if the data isn't available.

As you say, I think `map` is for situations where you a new array of the same length. If you want to create a new object (which may just mean an array of a different length because you've filtered some stuff, or may be a new data structure entirely) you can go up a level to `reduce`. In this case `reduce` is able to combine `map` and `filter` into one operation:

```ts
const enhanceCards = (collections: FEFrontCard[]): DCRFrontCard[] =>
	collections.reduce((enhanced: DCRFrontCard[], faciaCard: FEFrontCard) => {
		if (faciaCard.format) {
			const format = decideFormat(faciaCard.format);
			return [...enhanced, {
				format,
				isLiveBlog: format.design === ArticleDesign.LiveBlog,
				url: faciaCard.header.url,
				headline: faciaCard.header.headline,
				webPublicationDate: faciaCard.card.webPublicationDateOption
					? new Date(
							faciaCard.card.webPublicationDateOption,
					  ).toISOString()
					: undefined,
				image: faciaCard.properties.maybeContent?.trail.trailPicture
					?.allImages[0].url,
				kickerText:
					faciaCard.header.kicker?.item?.properties.kickerText,
			}];
		}

		return enhanced;
	}, []);
```

I think the code is similar to `forEach` without, as Max mentions, switching to mutation.

Reply:
Proposed as a simplification in #4582